### PR TITLE
Link Infiltration to HVAC Operation and Add Return Air Flow Controls

### PR DESCRIPTION
**Pull request by mjwitte**
Add new fields to ZoneHVAC:EquipmentConfiguration to allow optional control of zone return air flow rate.  Enhance ZoneAirMassFlowConservation to separately select whether mixing and/or infiltration will be used for balancing, and to allow infiltration balancing in any zone (not just mixing source zones).
- [x] Add new ZoneAirMassFlowConservation inputs
- [x] Implement new ZoneAirMassFlowConservation inputs
- [x] Add new return air inputs to ZoneHVAC:EquipmentConnections
- [x] Implement return air adjustments
- [x] Integrate with zone air mass balance
- [x] Fix infiltration heat balance disconnect
- [x] Example file
- [x] Save eio output changes for last
- [x] Address #4509 at the same time - eio alignment issue
- [x] Address #5042 zone mixing report variables
- [x] Unit tests
- [x] I/O Ref
- [x] Engineering Ref

### CODE REVIEW

**Comment by mjwitte** [comment]
@Myoldmopar It is unfortunate that the entire design doc lands at the top of the diffs page.  May want to reconsider the "design" folder name to put it at the bottom?

**Comment by Myoldmopar** [comment]
I am open to suggestion on removing it, although I'd say I personally like that as the heading that frames the whole topic.  The links are there in the "changed files" to jump to specific files.  Anyway, sure, I am fine with changing or whatever.

**Comment by mjwitte** [comment]
@Myoldmopar Doh!  Never knew those links were there.  Just never bothered to click on "n changed files".  Never mind.

**Comment by mjwitte** [comment]
@Myoldmopar Because this modifies the eio output line for Zone Air Mass Flow Balance Simulation it will cause failures in every file. Current:

```
! <Zone Air Mass Flow Balance Simulation>, Simulation {Yes/No}
 Zone Air Mass Flow Balance Simulation, No
```

Proposed new output for files that don't have this object (after I fix it):

```
! <Zone Air Mass Flow Balance Simulation>, Enforce Mass Balance {Yes | No}, Adjust Zone Mixing {Yes | No}, Adjust Zone Infiltration {AddInfiltration | AdjustInfiltration | NoInfiltration}, Infiltration Zones {MixingSourceZonesOnly | AllZones}
+ Zone Air Mass Flow Balance Simulation, No,N/A,N/A,N/A
```

How have we handled this in the past to isolate the eio output diffs from important diffs?

**Comment by Myoldmopar** [comment]
@mjwitte I'll run the suite locally to isolate out those diffs.  We could consider eio changes as a warning...

**Comment by mjwitte** [comment]
@Myoldmopar OK.  Also, I noticed these are all showing a table diff for the version.  e.g. see here
http://energyplus.s3-website-us-east-1.amazonaws.com/regressions/2015-08/b5d65e9c71fc9f1db7dda067c865b6fde819d5f9-96fe0b994247757763bfd8e6a2e4f6b4349dac93/1ZoneEvapCooler/Win64-Windows-7-VisualStudio-12/eplustbl.htm.absdiff.htm

Would be nice to teach tablediff to ignore that.

**Comment by mjwitte** [comment]
@Myoldmopar Realized that this will require a transition for ZoneAirMassFlowConservation because the new inputs decouple mixing and infiltration balance.  Where do I put the rules?

**Comment by Myoldmopar** [comment]
- Repo: EnergyPlusBuildSupport
- Branch: v8.4.0
- File: Rules830-840.xls or whatever the filename is...

**Comment by mjwitte** [comment]
@Myoldmopar Rules added in BuildSupport.

**Comment by Myoldmopar** [comment]
@mjwitte Updated transition.  Ran it with 5ZoneAirCooled_ZoneAirMassFlowBalance.idf.  Here is the diff:

```
ZoneAirMassFlowConservation,
    Yes,                     !- Adjust Zone Mixing For Zone Air Mass Flow Balance
    AddInfiltrationFlow;     !- Source Zone Infiltration Treatment
```

to 

```
  ZoneAirMassFlowConservation,
    Yes,                     !- Adjust Zone Mixing For Zone Air Mass Flow Balance
    AddInfiltrationFlow,     !- Infiltration Balancing Method
    MixingSourceZonesOnly;   !- Infiltration Balancing Zones
```

That look right to you?

**Comment by mjwitte** [comment]
@Myoldmopar Yes, thanks.  Also please test a copy where Adjust Zone Mixing = No.  Second field should change to None for that case.

**Comment by Myoldmopar** [comment]
```
ZoneAirMassFlowConservation,
    No,                     !- Adjust Zone Mixing For Zone Air Mass Flow Balance
    AddInfiltrationFlow;     !- Source Zone Infiltration Treatment
```

went to:

```
ZoneAirMassFlowConservation,
    No,                      !- Adjust Zone Mixing For Zone Air Mass Flow Balance
    None,                    !- Infiltration Balancing Method
    MixingSourceZonesOnly;   !- Infiltration Balancing Zones
```

**Comment by mjwitte** [comment]
Perfect!

**Comment by Myoldmopar** [comment]
:100: :+1:

**Comment by Myoldmopar** [comment]
Oh, and I didn't update the existing idfs with the new transition changes.  Do you need that or are you planning on doing it?

**Comment by mjwitte** [comment]
There's just one.  I'll do that.  And I need a new example file as well.  Or maybe I could modify this one.

File: src/EnergyPlus/SimAirServingZones.cc [hunk 1]
```diff
@@ -1959,12 +1959,6 @@ namespace SimAirServingZones {
 
 			} // end loop over outlet branches
 
-			// sum and save the total loop return air mass flow rate
-			AirLoopFlow( AirLoopNum ).TotReturn = 0.0;
-			for ( InBranchNum = 1; InBranchNum <= AirToZoneNodeInfo( AirLoopNum ).NumReturnNodes; ++InBranchNum ) {
-				AirLoopFlow( AirLoopNum ).TotReturn += Node( AirToZoneNodeInfo( AirLoopNum ).AirLoopReturnNodeNum( InBranchNum ) ).MassFlowRate;
-			}
-
```
Comment (mjwitte) [inline src/EnergyPlus/SimAirServingZones.cc:1994]: Note:  AirLoopFlow.TotReturn is never used anywhere else, so deleting this block.

**Comment by nmerket** [comment]
@jonwinkler You might find this interesting. Here's the [NFP](https://github.com/NREL/EnergyPlus/blob/81211464-LinkInfiltrationAndReturnFlows/design/FY2015/LinkInfiltrationToHVACOperation.md).

**Comment by Myoldmopar** [comment]
Ugh.  OK.  I got the merge conflict worked out.  They way develop was, the whole file was conflicted, so it was bad.  But once I merged in that hot-fix to repair the line endings in develop, it was much less work.  Basically just choosing develop for all the unrelated diffs and choosing yours for all the infiltration, etc., related diffs.  I guess this is (nearly?) good to go now?

**Comment by mjwitte** [comment]
@Myoldmopar Thanks.  I'll let you know when I've pushed up the last changes.

**Comment by Myoldmopar** [comment]
This is looking much better with that last commit.  Only the one diff file.  I see you've checked all the checkboxes above, but I don't see any additional commits.  Is this PR now fully flushed out and ready for final review/merge?

**Comment by mjwitte** [comment]
No, almost ready to push up today's changes which will cause _every_ file to have an eio diff.  I changed how the output variables get set up and now I'm getting zone air mass balance variables for the plenum which shouldn't be included.  Actually, I'm going to make a follow-up issue for that and call this done.

**Comment by Myoldmopar** [comment]
> No, almost ready to push up today's changes
> 
> Actually, I'm going to make a follow-up issue for that and call this done.

Commit e3c43eb

...ok so, you are done? :8ball: :smile:

**Comment by mjwitte** [comment]
Yes, sir!  Finished.

**Comment by Myoldmopar** [comment]
OK everyone listening; I'm merging this in, so pull in develop if you make any more commits because this will turn your CI result comments into a Tarantino film with all the eio diffs.

**Comment by macumber** [comment]
@mjwitte apparently I missed the boat on this but I am curious why we still need the 'Adjust Zone Mixing For Zone Air Mass Flow Balance' field, especially now that 'None' is an option for 'Infiltration Balancing Method'?  If you set 'Adjust Zone Mixing For Zone Air Mass Flow Balance' = 'Yes' and 'Infiltration Balancing Method' = 'None', what happens?  It seems like this object has one too many knobs.

**Comment by mjwitte** [comment]
@macumber The two are independent now. You can adjust only mixing, only infiltration, both, or none.

**Comment by macumber** [comment]
Oh I see, one adjusts mixing object and the other adjusts infiltration objects?  If you set 'Adjust Zone Mixing For Zone Air Mass Flow Balance' = True does that prevent a zone from receiving more air than it exhausts?

**Comment by mjwitte** [comment]
Correct.  It gets a little more complicated than just exhaust flow, but yes, it attempts to adjust incoming mixing flows up or down in order to balance the receiving zone.

**Pull request merged**